### ISSUE TITLE
Add EML feature to the SMTP analyzer

### ISSUE THREAD

**Issue by zrobinette12** [zeek/zeek#3900]
I have a use-case from Zeek to extract SMTP DATA as a file (a.k.a EML file type). EML files are undecoded SMTP MIME DATA. My organization developed a plugin that achieves this, however it's a copy pasta of the SMTP analyzer. We do this by removing the MIME_Mail analyzer from being called by the SMTP analyzer, and then do the following:

- At `ProcessData` https://github.com/zeek/zeek/blob/master/src/analyzer/protocol/smtp/SMTP.cc#L852 we'll forward the SMTP data to the file_mgr if we've reached a configurable buffer (which I think we just keep to 100mb or something).
- At `EndData` https://github.com/zeek/zeek/blob/master/src/analyzer/protocol/smtp/SMTP.cc#L900-L918 instead of doing `mail->Done()`, we also send that to the file_mgr.

Here's a specific example
```
void EML_Analyzer::EndData() {
	if (file_buf.length() > 0) {
		file_id = zeek::file_mgr->DataIn(reinterpret_cast<const u_char*>(file_buf.c_str()),
					   file_buf.length(), GetAnalyzerTag(), Conn(),
					   orig_is_sender, file_id);
		file_buf.clear();
	}

	zeek::file_mgr->EndOfFile(GetAnalyzerTag(), Conn());
	data_len = 0;
	file_id.clear();
}
```

This sends the undecoded MIME data to the file analysis framework (receives a fuid). This then lets us extract this data as a single file.

I'm having trouble maintaining this plugin and feel it's not a good solution since I'm making Zeek perform SMTP analysis twice (since of course I have other people who want the smtp.log and just the file attachments extracted as files).

TLDR:
Have some sort of config option that allows bypassing the MIME analyzer and treat SMTP DATA as a file.
